File: frontend/src/api/health.ts
/**
 * api/health.ts
 * Typed wrapper for the GET /api/health endpoint.
 */

export interface HealthResponse {
  status: 'ok' | string
  service: string
  version: string
  database: 'connected' | string
}

export async function checkHealth(): Promise<HealthResponse> {
  const res = await fetch('/api/health')
  if (!res.ok) {
    throw new Error(`Health check failed: ${res.status} ${res.statusText}`)
  }
  return res.json() as Promise<HealthResponse>
}

File: backend/routers/health.py
"""
routers/health.py — /api/health endpoint.
"""
from fastapi import APIRouter, Depends
from sqlalchemy.orm import Session
from sqlalchemy import text

from database import get_db

router = APIRouter(prefix="/api", tags=["infra"])


@router.get("/health")
def health_check(db: Session = Depends(get_db)) -> dict:
    """
    Returns 200 with DB connectivity status.
    Frontend polls this on load to confirm the stack is up.
    """
    try:
        db.execute(text("SELECT 1"))
        db_status = "connected"
    except Exception as exc:
        db_status = f"error: {exc}"

    return {
        "status": "ok",
        "service": "Polaris Voice Assistant",
        "version": "0.1.0",
        "database": db_status,
    }
